lib/Pool.js

```diff
@@ -103,7 +104,8 @@ class Pool {
       // If the number of connections waiting is more (shouldn't happen) or
       // equal to the number of connections connecting, then we will need to
       // create MORE connections.
-      if (this.connectingCount <= this.connectionQueue.length)
+      if (this.connectingCount <= this.connectionQueue.length &&
```

Comment:
Might have to ensure that maxSize is set (by default it is set to null, but maybe it can be set to `MAX_SAFE_INTEGER` or something). Also need to ensure that poolSize + incrementSize <= maxSize. Possible that increment size will make it go too far over. I think something like:
```
if (
        this.connectingCount <= this.connectionQueue.length &&
        (this.maxSize && this.maxSize > this.poolSize)
        )
      {
        let numNewConnections = this.incrementSize;
        // its possible that this.incrementSize will exceed this.maxSize number
        // of connections. Need to make sure that we don't break that limit.
        if (this.maxSize) {
          numNewConnections = min(this.incrementSize, this.maxSize - this.poolSize)
        }
        this.increasePoolSize(numNewConnections);
      }
```

Reply:
Good catch! Will do.